pkg/registry/core/pod/storage/eviction.go

```diff
@@ -307,12 +310,25 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 	return &metav1.Status{Status: metav1.StatusSuccess}, nil
 }
 
-func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string, validation rest.ValidateObjectFunc, options *metav1.DeleteOptions) error {
-	if feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
-		pod, err := getPod(r, ctx, name)
-		if err != nil {
-			return err
+func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string, pod *api.Pod, deleteOptions *metav1.DeleteOptions) error {
+	if !dryrun.IsDryRun(deleteOptions.DryRun) && feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		setPreconditions := func(_ context.Context, _, oldObj runtime.Object) (runtime.Object, error) {
+			// Throwaway the newObj. We care only about the latest pod obtained from etcd (oldObj).
+			// So we can add DisruptionTarget condition in conditionAppender without conflicts.
+			oldPodCopy := oldObj.(*api.Pod).DeepCopy()
+			if deleteOptions.Preconditions != nil {
+				// Insert preconditions into the pod object if we or the user specifically requested them,
+				// and let the update fail in case they do not pass.
+				if uid := deleteOptions.Preconditions.UID; uid != nil && len(*uid) > 0 {
+					oldPodCopy.UID = *uid
+				}
+				if rv := deleteOptions.Preconditions.ResourceVersion; rv != nil && len(*rv) > 0 {
```

Comment:
As above, wondering if `len(*rv) > 0)` is needed here, and if there isn't another layer that validates it.

Reply:
there is not, see above comment for why we should keep this (setting a "" resourceVersion actually makes the pod write less safe)